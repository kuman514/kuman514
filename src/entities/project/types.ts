export interface ProjectItem {
  projectId: string;
  title: string;
  thumbnailUrl: string;
  description: string;
  techStackTags: string[];
}
