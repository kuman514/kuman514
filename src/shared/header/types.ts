export interface NavNode {
  id: string;
  label: string;
  href: string;
}
