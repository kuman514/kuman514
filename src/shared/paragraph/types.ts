import { ReactNode } from 'react';

export interface ParagraphProps {
  customClassName?: string;
  children?: ReactNode;
}
