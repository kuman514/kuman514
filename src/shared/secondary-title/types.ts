import { ReactNode } from 'react';

export interface SecondaryTitleProps {
  customClassName?: string;
  children?: ReactNode;
}
