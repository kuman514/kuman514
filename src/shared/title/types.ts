import { ReactNode } from 'react';

export interface TitleProps {
  customClassName?: string;
  children?: ReactNode;
}
