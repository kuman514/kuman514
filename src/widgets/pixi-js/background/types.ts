import { Sprite } from 'pixi.js';

export interface IconComet {
  sprite: Sprite;
  scale: number;
  speed: number;
}
