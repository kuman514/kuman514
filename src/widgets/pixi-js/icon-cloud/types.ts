import { Sprite } from 'pixi.js';

export interface IconCloudIcon {
  sprite: Sprite;
  x: number;
  y: number;
  z: number;
}
