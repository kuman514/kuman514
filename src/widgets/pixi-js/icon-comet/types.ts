import { Sprite } from 'pixi.js';

export interface IconComet {
  sprite: Sprite;
  scale: number;
  speed: number;
}

export interface CanvasSize {
  canvasWidth: number;
  canvasHeight: number;
}

export interface RandomizeIconCometPositionParams extends CanvasSize {
  icon: IconComet;
  isInitial: boolean;
}
