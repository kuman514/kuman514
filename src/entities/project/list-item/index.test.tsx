import { fireEvent, render, screen } from '@testing-library/react';

import { ProjectItem } from '../types';
import ProjectListItem from '.';

const testProjectItem: ProjectItem[] = [
  {
    projectId: 'Koishi',
    title: 'Koishi Komeiji',
    thumbnailUrl:
      'https://pbs.twimg.com/profile_images/1175746616545857538/JuT5t8dm.jpg',
    description: 'A Youkai girl of unconsciousness from Touhou Project.',
    techStackTags: [
      'Touhou Project',
      'Subterranean Animism',
      'Hopeless Masquerade',
      'Moshi Moshi Meridesu',
      'Unconsciousness',
      'Hartmann No Youkai Shoujo',
    ],
  },
  {
    projectId: 'Hoshino',
    title: 'Hoshino Takanashi',
    thumbnailUrl: 'https://pbs.twimg.com/media/FtlqZPhaAAAnEqZ.jpg?name=orig',
    description: 'A 3rd-grade high school student. Appears in Blue Archive.',
    techStackTags: [
      'Blue Archive',
      'Abydos High School',
      'Shot Gun',
      'Tank',
      'Foreclosure Task Force',
    ],
  },
];

const mockFn = jest.fn();

function MockedLink({
  children,
  href,
  'aria-label': ariaLabel,
}: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
  return (
    <a
      aria-label={ariaLabel}
      onClick={() => {
        mockFn(href);
      }}
    >
      {children}
    </a>
  );
}

jest.mock('next/link', () => MockedLink);

describe('ProjectListItem', () => {
  it('should have appearances that matches snapshot', () => {
    const { container } = render(
      testProjectItem.map((item) => (
        <ProjectListItem key={`project-${item.projectId}`} projectItem={item} />
      ))
    );
    expect(container).toMatchSnapshot();
  });

  it('should have nodes that have working links which takes you to /projects/:projectId', async () => {
    render(
      testProjectItem.map((item) => (
        <ProjectListItem key={`project-${item.projectId}`} projectItem={item} />
      ))
    );

    fireEvent.click(await screen.findByText(/Koishi Komeiji/i));
    expect(mockFn).toHaveBeenLastCalledWith('/projects/Koishi');

    fireEvent.click(await screen.findByText(/Hoshino Takanashi/i));
    expect(mockFn).toHaveBeenLastCalledWith('/projects/Hoshino');
  });
});
