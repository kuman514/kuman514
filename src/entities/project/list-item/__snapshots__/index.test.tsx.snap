// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListItem should have appearances that matches snapshot 1`] = `
<div>
  <li
    class="border-b border-b-gray-400 last:border-b-0 transition hover:bg-green-400/[.5]"
  >
    <a>
      <img
        alt="Koishi Komeiji - A Youkai girl of unconsciousness from Touhou Project."
        class="w-full aspect-3/2 object-cover"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1175746616545857538%2FJuT5t8dm.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1175746616545857538%2FJuT5t8dm.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1175746616545857538%2FJuT5t8dm.jpg&w=3840&q=75 2x"
        style="color: transparent;"
        width="1500"
      />
      <div
        class="flex flex-col justify-center gap-1 md:py-3"
      >
        <span
          class="font-bold text-2xl md:text-3xl"
        >
          Koishi Komeiji
        </span>
        <span
          class="pb-3"
        >
          A Youkai girl of unconsciousness from Touhou Project.
        </span>
        <div
          class="flex flex-row flex-wrap gap-1"
        >
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Touhou Project
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Subterranean Animism
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Hopeless Masquerade
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Moshi Moshi Meridesu
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Unconsciousness
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Hartmann No Youkai Shoujo
          </span>
        </div>
      </div>
    </a>
  </li>
  <li
    class="border-b border-b-gray-400 last:border-b-0 transition hover:bg-green-400/[.5]"
  >
    <a>
      <img
        alt="Hoshino Takanashi - A 3rd-grade high school student. Appears in Blue Archive."
        class="w-full aspect-3/2 object-cover"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fmedia%2FFtlqZPhaAAAnEqZ.jpg%3Fname%3Dorig&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fmedia%2FFtlqZPhaAAAnEqZ.jpg%3Fname%3Dorig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fmedia%2FFtlqZPhaAAAnEqZ.jpg%3Fname%3Dorig&w=3840&q=75 2x"
        style="color: transparent;"
        width="1500"
      />
      <div
        class="flex flex-col justify-center gap-1 md:py-3"
      >
        <span
          class="font-bold text-2xl md:text-3xl"
        >
          Hoshino Takanashi
        </span>
        <span
          class="pb-3"
        >
          A 3rd-grade high school student. Appears in Blue Archive.
        </span>
        <div
          class="flex flex-row flex-wrap gap-1"
        >
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Blue Archive
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Abydos High School
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Shot Gun
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Tank
          </span>
          <span
            class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
          >
            Foreclosure Task Force
          </span>
        </div>
      </div>
    </a>
  </li>
</div>
`;
